tests/minifier.js

```diff
@@ -840,6 +840,54 @@
     equal(minify(input, { removeOptionalTags: true }), output);
     equal(minify(input, { removeOptionalTags: true, removeEmptyElements: true }), output);
 
+    input = ' <html></html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html> </html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html></html> ';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = ' <html><body></body></html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html> <body></body></html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body> </body></html>';
+    output = '<body> ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body></body> </html>';
+    output = ' ';
+    equal(minify(input, { removeOptionalTags: true }), output);
+    output = '';
+    equal(minify(input, { collapseWhitespace: true, removeOptionalTags: true }), output);
+
+    input = '<html><body></body></html> ';
+    output = ' ';
```

Comment:
These tests are for the benefit of keeping track of this tricky logic than of any real world importance, I agree.

Pretty sure most people can't tell an empty screen from one with a whitespace on it :ghost: